### PR TITLE
Migrate to gomod

### PR DESCRIPTION
**Pull request by spolcyn**
Changed directory structure to https://github.com/golang-standards/project-layout, migrated to gomod, and fixed incompatibility with changes in Prometheus components.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -31,9 +31,7 @@ version.properties:
 build: $(TARGET)
 
 $(TARGET): .target_os $(SOURCES)
-	$(if $(shell command -v dep 2> /dev/null),$(info Found golang/dep),$(error Please install golang/dep))
-	dep ensure
-	GOOS=$(OS) GOARCH=${ARCH} CGO_ENABLED=0 go build -a -installsuffix cgo --ldflags '-w' -o $@ 
+	GOOS=$(OS) GOARCH=${ARCH} CGO_ENABLED=0 go build -a --ldflags '-w' -o $@ ./cmd/$@
```
Comment (RobAtticus) [inline Makefile:34]: does `make test` still work? just want to make sure the macros above don't need to be changed

**Review by RobAtticus — APPROVED** [review]
Awesome, just one quick check but then it can be merged

**Comment by RobAtticus** [comment]
Looks like Travis config needs updating too. Let's make it consistent with our other repos actually:

See:
https://github.com/timescale/timescaledb-tune/blob/master/.coverage_tests.sh
and
https://github.com/timescale/timescaledb-tune/blob/master/.travis.yml

In particular we should def test Go 1.11 and 1.12 (travis.yml) and coverage testing would be nice.

File: Makefile (hunk 1 above)
Comment (spolcyn) [reply Makefile:34]: Works locally + is part of the travis tests

**Pull request merged**